jib-cli/src/main/java/com/google/cloud/tools/jib/cli/war/WarFiles.java

```diff
@@ -19,30 +19,75 @@
 import com.google.cloud.tools.jib.api.InvalidImageReferenceException;
 import com.google.cloud.tools.jib.api.Jib;
 import com.google.cloud.tools.jib.api.JibContainerBuilder;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
 import com.google.cloud.tools.jib.cli.ArtifactProcessor;
+import com.google.cloud.tools.jib.cli.CommonCliOptions;
+import com.google.cloud.tools.jib.cli.CommonContainerConfigCliOptions;
+import com.google.cloud.tools.jib.cli.ContainerBuilders;
+import com.google.cloud.tools.jib.plugins.common.logging.ConsoleLogger;
 import com.google.common.collect.ImmutableList;
 import java.io.IOException;
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
 public class WarFiles {
 
   /**
    * Generates a {@link JibContainerBuilder} from contents of a WAR file.
    *
    * @param processor artifact processor
+   * @param commonCliOptions common cli options
+   * @param commonContainerConfigCliOptions common cli options shared between jar and war command
+   * @param logger console logger
    * @return JibContainerBuilder
-   * @throws IOException if I/O error occurs when opening the war file or if temporary directory
+   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
    *     provided doesn't exist
    * @throws InvalidImageReferenceException if the base image reference is invalid
    */
-  public static JibContainerBuilder toJibContainerBuilder(ArtifactProcessor processor)
+  public static JibContainerBuilder toJibContainerBuilder(
+      ArtifactProcessor processor,
+      CommonCliOptions commonCliOptions,
+      CommonContainerConfigCliOptions commonContainerConfigCliOptions,
+      ConsoleLogger logger)
       throws IOException, InvalidImageReferenceException {
-    JibContainerBuilder containerBuilder = Jib.from("jetty");
+    JibContainerBuilder containerBuilder;
+    List<FileEntriesLayer> layers;
+    Optional<String> baseImage = commonContainerConfigCliOptions.getFrom();
+    if (baseImage.isPresent()) {
+      containerBuilder =
+          ContainerBuilders.create(
+              commonContainerConfigCliOptions.getFrom().get(),
+              Collections.emptySet(),
+              commonCliOptions,
+              logger);
+    } else {
+      containerBuilder = Jib.from("jetty");
```

Comment:
This makes me wonder why the other branch can't juse be `Jib.from(baseImage.get())` or why this branch doesn't use `ContainerBuilders.create("jetty", ...)`.

Reply:
`ContainerBuilders.create("jetty")` can be used in place of `Jib.from` but the difference is that the former does some extra credential work to get credentials for a base image registry while the latter doesn't. I assumed that this may not be needed if we were to use the default? This way we don't have to log the following messaging every time:

```
[WARN] The credential helper (docker-credential-desktop) has nothing for server URL: registry-1.docker.io
Got output:
credentials not found in native keychain
```

Reply:
Hmm... I am aware of the ugliness of the warning (which the user can resolve by configuring `~/.docker/config.json` in a more modern way). However, I think we should still configure auth for public Docker Hub images. Logging in increases the rate limit.

Reply:
Oh that is a really good point. Modified this to use `ContainerBuilders.create("jetty")`. instead of `Jib.from()`